### ISSUE TITLE
Understanding Timestamp Feature

### ISSUE THREAD

**Issue by turnerneil** [tensorflow/recommenders#230]
I'm interested in learning about the timestamp features in the guides.

1. In the Deep model tutorial we pass user_id and timestamp into the query model. According to the dataset definition, "the timestamp of the ratings, represented in seconds since midnight Coordinated Universal Time (UTC) of January 1, 1970". This means that query input at prediction time will have timestamps that are not in the training set. Assuming query input is an existing user_id, will the trained model make a prediction for the user_id and alter it slightly based on the timestamp? 
2. The Leveraging context features guide uses day of week as example use case for timestamps: "For example, day of the week may be an important feature when deciding whether to recommend a short clip or a movie". Does this mean the timestamp embedding represent latent features such as month, day of week, and hour of day? 
3. What happens when you add a feature in the user model, but don't specify it as a query input(ex: normalized timestamps)? Does this feature help train the model on existing user_ids, but since it's not in the query input it wont have any effect if predicting recommendations for a new user_id?

Any additional insight about the using timestamps in a deep retrieval model would be appreciated. Thanks!

**Comment by maciejkula** [comment]
One: The use of timestamp embeddings in the tutorial is mostly a teaching device, and it will likely not work in a real setting, for the reason you point you. The tutorial says:

```
Note that our use of timestamp features in this tutorial interacts with our choice of training-test split in an undesirable way. Because we have split our data randomly rather than chronologically (to ensure that events that belong to the test dataset happen later than those in the training set), our model can effectively learn from the future. This is unrealistic: after all, we cannot train a model today on data from tomorrow.

This means that adding time features to the model lets it learn future interaction patterns. We do this for illustration purposes only: the MovieLens dataset itself is very dense, and unlike many real-world datasets does not benefit greatly from features beyond user ids and movie titles.
```
Two: what you'd do instead is have features that contain day of week/month etc, and embed those. They will be more useful in prediction because they will hopefully capture stable cyclical patterns.

3. That's possible, depending on how you set up your model. In the common case you have to have the same features in training and serving.

**Comment by Flipper-afk** [comment]
2. There's a nice approach in the [TF Time series tutorial](https://www.tensorflow.org/tutorials/structured_data/time_series#time) (scroll a bit down), using `sin` and `cos`, you can try to convert the timeline to a "clock".
